### PR TITLE
Jb refactor filter

### PR DESCRIPTION
**Pull request by jbevis**
### What's this PR do?
I pulled the logic for the difficulty filter out of the card grid component and moved it into the displayReducer. As such, the filter reducer became irrelevant and so I removed that and the set filter action out right. Instead of returning all the trails if none match the filter, the component renders a new section with the controls so users can perform another search.

### Pre-Merge TODOs
- [x] Tests Added
- [ ] Needs 2 peer reviews
- [x] Add follow up issues
- 
### Learning
While pairing with Meeka, she was able to discern that the filter reducer was not necessary. So we were able to pull the logic out and refactor the function to work within the confines of the display reducer.

### Where should the reviewer start?
displayReducer.js
CardGrid.js

### Any background context you want to provide?
While I added a check earlier just to return all the trails if the difficulty didn't match any trails. This seems to operate cleaner and removes unnecessary code. I will also need to make the app more modular, there is too much repetitive code, and can be broken into new components.

### What gif best describes this PR or how it makes you feel?
[This one](https://media.giphy.com/media/uN0AD5n8NjYqc/giphy.gif)

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -195,7 +207,7 @@ button:hover {
 }
 
 .trail-conditions p {
-  max-width: 80px;
+  max-width: 300px;
```
Comment (JohnBinning) [inline src/index.css:210]: Really nice setting of width here.  I expected no less.

File: src/reducers/displayReducer.js [hunk 2]
```diff
@@ -1,10 +1,25 @@
+const filterByDifficulty = (trails, diffLevel) => {
+  return Object.keys(trails).reduce((acc, key) => {
+    if (trails[key].difficulty === diffLevel) {
+      acc[key] = trails[key]
+    } else if (diffLevel === 'show-all') {
+      return trails
+    }
+    return acc
+  }, {});
```
Comment (JohnBinning) [inline src/reducers/displayReducer.js:9]: is that a rogue semicolon?

File: src/index.css [hunk 3]
```diff
@@ -136,6 +140,15 @@ button:hover {
   margin: 10px 30px;
 }
 
+#back-link {
+  font-family: 'Open Sans', sans-serif;
+  cursor: pointer;
+}
+
+#back-link:hover {
+  color: #E6DED1
```
Comment (JohnBinning) [inline src/index.css:149]: missing a semicolon.

File: src/components/CardGrid/CardGrid.test.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ describe('CardGrid testing', () => {
     filterSet.simulate('change', { target: { value: 'greenBlue' }})
     filterBtn.simulate('click')
 
-    expect(mockFn).toHaveBeenCalledTimes(2)
+    expect(mockFn).toHaveBeenCalledTimes(1)
```
Comment (JohnBinning) [inline src/components/CardGrid/CardGrid.test.js:41]: perhaps give filterBtn it's own unique mock function (I saw you passed mockFn to three different props).  Then you can be more certain that the filter mock function got called.  If you want to get really crazy, you could use .toHaveBeenCalledWith to test that it is getting the correct args as well.

**Comment by JohnBinning** [comment]
Solid reorganization.  The app is looking really good, especially with all the new widths you've set.

File: src/components/CardGrid/CardGrid.js [hunk 5]
```diff
@@ -10,29 +11,44 @@ export default class CardGrid extends Component {
...
+          </div>
+          <section className='welcome-card'>
+            <h3>Sorry, no trails matched that difficulty</h3>
+            <p id='back-link'
+               onClick={() => handleSetDisplay(trails, 'show-all') } >
+              Back to trails
+            </p>
+          </section>
+        </section>
       )
```
Comment (devthehuman) [inline src/components/CardGrid/CardGrid.js]: This is a bit long and could be harder to read, maybe break it out into it's own dumb component that you could test as well?

File: src/components/CardGrid/CardGrid.js [hunk 6]
```diff
@@ -10,29 +11,44 @@ export default class CardGrid extends Component {
...
+            </button>
+          </div>
+          <section className='welcome-card'>
+            <h3>Sorry, no trails matched that difficulty</h3>
+            <p id='back-link'
+               onClick={() => handleSetDisplay(trails, 'show-all') } >
+              Back to trails
+            </p>
+          </section>
+        </section>
```
Comment (ghost) [inline src/components/CardGrid/CardGrid.js:48]: This is a bit long and could be harder to read, maybe break it out into it's own dumb component that you could test as well?

File: src/index.css [hunk 7]
```diff
@@ -123,6 +123,10 @@ button:hover {
   max-width: 80%;
 }
 
+.card-grid label {
+  color: #FFE242;
+}
+
 #filter-bar {
   margin: 10px;
```
Comment (ghost) [inline src/index.css:131]: You can remove this line since you're setting the margin on line 140 as well.

File: src/reducers/reducers.test.js [hunk 8]
```diff
@@ -1,6 +1,5 @@
 import conditionsReducer from './conditionsReducer';
 import displayReducer from './displayReducer';
-import filterReducer from './filterReducer';
 import trailsReducer from './trailsReducer';
 
 import expect from 'expect';
```
Comment (ghost) [inline src/reducers/reducers.test.js:5]: I had no idea you could import this, that's super cool.

**Comment by ghost** [comment]
Overall, great refactor dude!

**Comment by jbevis** [comment]
Awesome feedback, thanks guys.

To your point Dev, yeah the CardGrid is way too long now, I plan on breaking it out into a few more dumb components.

@joshuajhun, @Tman22, @rrgayhart for review.

**Pull request merged**